.github/workflows/workflow.yml

```diff
@@ -1,19 +1,29 @@
 name: main
 on:
-  push: { branches: [main] }
-  pull_request: { branches: [main] }
+  push: 
+    branches: [main]
+    tags: "v*"
+  pull_request:
+    branches: [main]
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
+env:
+  BUILD_DEB_OUTPUT_DIR: out
+  BUILD_PYTHON_OUTPUT_DIR: dist
+  # The latest version supporting Python 3.6
+  POETRY_VERSION: 1.1.15
+
 
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-python@v4
-      with: { python-version: "3.11", cache: pip }
-    - run: pip install .[test]
+    - uses: actions/checkout@v3                
+    - uses: ./.github/actions/setup_dependencies
+      with:
+        python-version: "3.10.12"
```

Comment:
Wouldn't it better to pin minor version rather then patch version? (e.g. use `3.10` instead of `3.10.12`)

Reply:
Done